builtin/logical/transit/path_encrypt.go

```diff
@@ -158,6 +175,145 @@ func (b *backend) pathEncryptWrite(
 		return logical.ErrorResponse("policy not found"), logical.ErrInvalidRequest
 	}
 
+	batchInputRaw := d.Get("batch").(string)
+	var batchInput []byte
+	if len(batchInputRaw) != 0 {
+		batchInput, err = base64.StdEncoding.DecodeString(batchInputRaw)
+		if err != nil {
+			return logical.ErrorResponse("failed to base64-decode batch"), logical.ErrInvalidRequest
+		}
+	}
+
+	if len(batchInput) != 0 {
```

Comment:
I am almost done through this change. But I have some thoughts here. The reason I had shied away from this was to not cause any BC issues. The way errors are handled for batch requests are different from the way we handle errors for normal case. Specifically the error after p.Encrypt(), where we distinguish UserError and InternalError.

Do you still want me to push this up?